ssl/statem/statem.c

```diff
@@ -347,14 +347,16 @@ static int state_machine(SSL *s, int server)
                  */
                 s->ctx->stats.sess_accept_renegotiate++;
             }
+
+            s->s3->tmp.cert_request = 0;
         } else {
             s->ctx->stats.sess_connect++;
 
             /* mark client_random uninitialized */
             memset(s->s3->client_random, 0, sizeof(s->s3->client_random));
             s->hit = 0;
 
-            s->s3->tmp.cert_request = 0;
+            s->s3->tmp.cert_req = 0;
```

Comment:
Side note: we should consider renaming `cert_req` or `cert_request`.  I cannot easily see how one differs from the other except by name.